tests/Keycloak/TokenStrategy/ClientCredentialsTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Keycloak\TokenStrategy;
+
+use App\Keycloak\Dto\Config;
+use App\Keycloak\Dto\Realm;
+use App\Keycloak\Exception\KeyCloakApiFailed;
+use App\Keycloak\TokenStrategy\ClientCredentials;
+use GuzzleHttp\Client;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+
+final class ClientCredentialsTest extends TestCase
+{
+    private Config $config;
+
+    protected function setUp(): void
+    {
+        $this->config = new Config(
+            true,
+            'https://keycloak.example.com/',
+            'php_client',
+            'a_true_secret',
+            new Realm('uitidpoc', 'Acceptance')
+        );
+    }
+
+    public function testHappyPath(): void
```

Comment:
Tests are snake_cased

Reply:
The name can also be more descriptive, something like `test_it_returns_a_valid_token`